amazon/src/features/auth/authSlice.ts

```diff
@@ -0,0 +1,146 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+
+import authService from './services/auth.service';
+import { NewUser } from './models/NewUser';
+import { DisplayUser } from './models/DisplayUser.interface';
+import { Jwt } from './models/Jwt';
+import { RootState } from '../../store';
+import { LoginUser } from './models/LoginUser.interface';
+
+const storedUser: string | null = localStorage.getItem('user');
+const user: DisplayUser | null = !!storedUser ? JSON.parse(storedUser) : null;
+
+const storedJwt: string | null = localStorage.getItem('jwt');
+const jwt: Jwt = !!storedJwt ? JSON.parse(storedJwt) : null;
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider refactoring user and jwt initialization</b></div>
<div id="fix">

Consider moving the initialization of `user` and `jwt` variables to a separate function for better readability and maintainability. This could also help with potential testing scenarios.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -10,5 +10,12 @@
 -const storedUser: string | null = localStorage.getItem('user');
 -const user: DisplayUser | null = !!storedUser ? JSON.parse(storedUser) : null;
 -
 -const storedJwt: string | null = localStorage.getItem('jwt');
 -const jwt: Jwt = !!storedJwt ? JSON.parse(storedJwt) : null;
 +const initializeAuthState = () => {
 +  const storedUser: string | null = localStorage.getItem('user');
 +  const user: DisplayUser | null = !!storedUser ? JSON.parse(storedUser) : null;
 +
 +  const storedJwt: string | null = localStorage.getItem('jwt');
 +  const jwt: Jwt = !!storedJwt ? JSON.parse(storedJwt) : null;
 +
 +  return { user, jwt };
 +};
 +
 +const { user, jwt } = initializeAuthState();
```
</div>
</details>
</div>

<div id="suggestion">
<div id="issue"><b>Consider simplifying boolean expressions</b></div>
<div id="fix">

It appears there are redundant double negations in the code. Consider simplifying the boolean expressions on lines 11 and 14. For example, `!!storedUser` can be replaced with `Boolean(storedUser)` or simply `storedUser != null`. This change would improve code readability without altering the logic.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -11,4 +11,4 @@
 -const user: DisplayUser | null = !!storedUser ? JSON.parse(storedUser) : null;
 +const user: DisplayUser | null = storedUser ? JSON.parse(storedUser) : null;

```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/omkardongre/amazon-clone/pull/5#issuecomment-2466913579>#601468</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged